### ISSUE TITLE
Implement MomentumIterativeGradient attack

### ISSUE THREAD

**Issue by wielandbrendel** [bethgelab/foolbox#138]
The winning attack in last years NIPS competition:

https://github.com/tensorflow/cleverhans/blob/538bf672faeeb355bc13e08f2916eb192a22e7e7/cleverhans/attacks.py#L462

Paper:
https://github.com/tensorflow/cleverhans/blob/538bf672faeeb355bc13e08f2916eb192a22e7e7/cleverhans/attacks.py#L462

We should implement this attack both for iterative FGSM as well as the standard Gradient descent. There seems to be already an implementation by a Foolbox user (but I haven't checked the code and compared to the paper):

https://github.com/neale/foolbox/blob/master/NIPS_attack/momentum_attack.py

**Comment by jonasrauber** [comment]
closed by #171
